Sources/SpeziConsent/Model/ConsentDocumentParser.swift

```diff
@@ -0,0 +1,590 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2025 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+// swiftlint:disable file_length
+
+import Foundation
+
+
+/// An error that occurred when parsing markdown text.
+public struct ConsentParseError: Error, Hashable {
```

Comment:
That's a huge file, might make sense to split up?

Reply:
don't think that'd make sense; it's large but its self-contained in the sense that all parsing-related functionality is in here (and as a result can be made `private` whereever possible).